session_subscribe_update_test.go

```diff
@@ -0,0 +1,579 @@
+		lt := &localTrack{
+			requestID:  42,
+			trackAlias: 100,
+		}
+		s.localTracks.addPending(lt)
+		// Don't confirm it - leave it in pending state
+		
+		// Should still find the track since findByID checks both pending and open
+		mh.EXPECT().enqueue(context.Background(), gomock.Any()).Return(nil)
+		
+		err := s.onSubscribeUpdate(&wire.SubscribeUpdateMessage{
+			RequestID:          42,
+			StartLocation:      wire.Location{Group: 10, Object: 5},
+			EndGroup:           20,
+			SubscriberPriority: 50,
+			Forward:            1,
+			Parameters:         wire.KVPList{},
+		})
+		
+		assert.NoError(t, err)
+	})
+
+	t.Run("receive_subscribe_update_message", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		cms := NewMockControlMessageSendQueue[wire.ControlMessage](ctrl)
+		mh := NewMockControlMessageRecvQueue[*Message](ctrl)
+		
+		s := newSession(cms, mh, ProtocolQUIC, PerspectiveServer)
+		
+		// Add a local track
+		lt := &localTrack{
+			requestID:  42,
+			trackAlias: 100,
+		}
+		s.localTracks.addPending(lt)
+		s.localTracks.confirm(42)
+		
+		// Mark handshake as done
+		close(s.handshakeDoneCh)
+		
+		// Expect the message to be passed to application
+		mh.EXPECT().enqueue(context.Background(), gomock.Any()).Return(nil)
+		
+		// Simulate receiving SUBSCRIBE_UPDATE via control stream
+		err := s.receive(&wire.SubscribeUpdateMessage{
+			RequestID:          42,
+			StartLocation:      wire.Location{Group: 10, Object: 5},
+			EndGroup:           20,
+			SubscriberPriority: 50,
+			Forward:            1,
+			Parameters:         wire.KVPList{},
+		})
+		
+		assert.NoError(t, err)
+	})
+
+	t.Run("subscribe_update_priority_values", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		cms := NewMockControlMessageSendQueue[wire.ControlMessage](ctrl)
+		mh := NewMockControlMessageRecvQueue[*Message](ctrl)
+		
+		s := newSession(cms, mh, ProtocolQUIC, PerspectiveServer)
+		
+		// Add a local track
+		lt := &localTrack{
+			requestID:  42,
+			trackAlias: 100,
+		}
+		s.localTracks.addPending(lt)
+		s.localTracks.confirm(42)
+		
+		// Test various priority values (0-255)
+		priorities := []uint8{0, 1, 127, 128, 255}
+		
+		for _, priority := range priorities {
+			mh.EXPECT().enqueue(context.Background(), &Message{
+				Method:             MessageSubscribeUpdate,
+				RequestID:          42,
+				TrackAlias:         100,
+				SubscriberPriority: priority,
+				StartGroup:         0,
+				StartObject:        0,
+				EndGroup:           0,
+				Forward:            1,
+			}).Return(nil)
+			
+			err := s.onSubscribeUpdate(&wire.SubscribeUpdateMessage{
+				RequestID:          42,
+				StartLocation:      wire.Location{Group: 0, Object: 0},
+				EndGroup:           0,
+				SubscriberPriority: priority,
+				Forward:            1,
+				Parameters:         wire.KVPList{},
+			})
+			
+			assert.NoError(t, err, "Priority %d should be accepted", priority)
+		}
+	})
+
+	t.Run("subscribe_update_forward_flag", func(t *testing.T) {
```

Comment:
Again, I don't think we need this case, since it is the same as above. It should be enough to test once. Also, I don't think forward is intended to be used as pause/resume? I thought it was just a signal for relays to prepare subscriptions, i.e., pull from upstream to be ready for a real subscription that might come later.